File: 771.jewels-and-stones.py
#
# @lc app=leetcode id=771 lang=python
#
# [771] Jewels and Stones
#

# @lc code=start
class Solution(object):
    def numJewelsInStones(self, jewels, stones):
        """
        :type jewels: str
        :type stones: str
        :rtype: int
        """
        j = set()
        for i in jewels:
            j.add(i)
        count = 0
        for i in stones:
            if i in j:
                count += 1
        return count
        
# @lc code=end

File: 771.jewels-and-stones.java
/*
 * @lc app=leetcode id=771 lang=java
 *
 * [771] Jewels and Stones
 */

// @lc code=start
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
            HashSet<Character> jewel = new HashSet<>();
    for (char c : jewels.toCharArray()) {
      jewel.add(c);
    }
    int count = 0;
    for (char s : stones.toCharArray()) {
      if (jewel.contains(s)) {
        count++;
      }
    }
    return count;
    }
}
// @lc code=end

